api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -11,88 +12,128 @@
   }
 
-  public void setOrganizationTypeOtherText(String organizationTypeOtherText) {
+  public DbInstitution setOrganizationTypeOtherText(String organizationTypeOtherText) {
     this.organizationTypeOtherText = organizationTypeOtherText;
+    return this;
   }
 
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+  @NotNull
   public Set<DbInstitutionEmailDomain> getEmailDomains() {
     return emailDomains;
   }
 
-  public void setEmailDomains(Iterable<DbInstitutionEmailDomain> emailDomains) {
-    this.emailDomains =
-        StreamSupport.stream(emailDomains.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Effectively: do an in-place this.emailDomains = emailDomains
+   *
+   * <p>Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First,
+   * call retainAll() to subset DB rows to those we wish to keep: the intersection of old and new.
+   * Then, call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailDomains the new set of domains for this Institution
+   */
+  public DbInstitution setEmailDomains(final Set<DbInstitutionEmailDomain> emailDomains) {
+    final Set<DbInstitutionEmailDomain> attachedDomains =
+        emailDomains.stream()
+            .map(domain -> domain.setInstitution(this))
+            .collect(Collectors.toSet());
+    // modifies this set so that its value is the intersection of the two sets
+    this.emailDomains.retainAll(attachedDomains);
+    this.emailDomains.addAll(Sets.difference(attachedDomains, this.emailDomains));
+    return this;
   }
 
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+  @NotNull
   public Set<DbInstitutionEmailAddress> getEmailAddresses() {
     return emailAddresses;
   }
 
-  public void setEmailAddresses(Iterable<DbInstitutionEmailAddress> emailAddresses) {
-    this.emailAddresses =
-        StreamSupport.stream(emailAddresses.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Effectively: do an in-place this.emailAddresses = emailAddresses
+   *
+   * <p>Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First,
+   * call retainAll() to subset DB rows to those we wish to keep: the intersection of old and new.
+   * Then, call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailAddresses the new set of addresses for this Institution
+   */
+  public DbInstitution setEmailAddresses(final Set<DbInstitutionEmailAddress> emailAddresses) {
+    final Set<DbInstitutionEmailAddress> attachedAddresses =
+        emailAddresses.stream()
+            .map(address -> address.setInstitution(this))
```

Comment:
I think we should validate that the incoming emailAddress doesn't already have a different institution. Otherwise this is destructive to that institution's map of the world.

Reply:
That is not how this works.  See the design doc.